system-parachains/encointer/src/migrations_fix.rs

```diff
@@ -17,173 +17,115 @@
 // the following are temporary local migration fixes to solve inconsistencies caused by not
 // migrating Storage at the time of migrating runtime code
 
-pub mod balances {
+pub mod collator_selection_init {
 	use frame_support::traits::OnRuntimeUpgrade;
-	use pallet_balances::*;
 	#[cfg(feature = "try-runtime")]
 	use sp_runtime::TryRuntimeError;
 
 	/// The log target.
-	const TARGET: &'static str = "runtime::fix::balances::migration";
-	pub mod v1 {
+	const TARGET: &'static str = "runtime::fix::collator_selection_init";
+	pub mod v0 {
 		use super::*;
-		use frame_support::pallet_prelude::*;
-		pub struct BruteForceToV1<T>(sp_std::marker::PhantomData<T>);
-
-		impl<T: Config> OnRuntimeUpgrade for BruteForceToV1<T> {
+		use codec::EncodeLike;
+
+		//crate::Pallet::ExistentialDeposit;
+		use crate::SessionKeys;
+		use frame_support::{pallet_prelude::*, traits::Currency};
+		use hex_literal::hex;
+		use log::info;
+		use parachains_common::impls::BalanceOf;
+		use sp_core::{crypto::key_types::AURA, sr25519};
+
+		const INVULNERABLE_A: [u8; 32] =
+			hex!("5e962096da68302d5c47fce0178d72fab503c4f00a3f1df64856748f0d9dd51e");
+		const INVULNERABLE_B: [u8; 32] =
+			hex!("0cecb8d1c2c744ca4c5cea57f5d6c40238f4dad17afa213672b8b7d43b80a659");
+		const INVULNERABLE_C: [u8; 32] =
+			hex!("ca1951a3c4e100fb5a899e7bae3ea124491930a72000c5e4b2775fea27ecf05d");
+		const INVULNERABLE_D: [u8; 32] =
+			hex!("484b443bd95068b860c92b0f66487b78f58234eca0f88e2adbe80bae4807b809");
+		const INVULNERABLE_E: [u8; 32] =
+			hex!("6c642fb4b571a5685a869cd291fafd575be47a918b231ba28165e5c0cd0cfa15");
+
+		pub struct InitInvulnerables<T>(sp_std::marker::PhantomData<T>);
+		impl<T> OnRuntimeUpgrade for InitInvulnerables<T>
```

Comment:
@joepetrowski may I ask you for  a sanity-check?
All we want to do is keep the same collators we have now as invulnerables and be sure to produce blocks after the runtime upgrade ;-)

Reply:
It looks like this sets everything needed. It's weird to me that the collators have the same Aura key and Account ID. @georgepisaltu could also do a sanity check on this.

Reply:
The `ValidatorId` happens to be the same type as `AccountId`. The key type is built with `SessionKeys { aura: sr25519::Public::from_raw(pk).into() }.into()`.

> All we want to do is keep the same collators we have now as invulnerables and be sure to produce blocks after the runtime upgrade

About this, I looked at the logic below and it seems you're setting all the storage values to what you want. I didn't check the hardcoded addresses though 😁 .